net: eth: Add native-sim support to native-posix Ethernet driver

jukkar opened the pull request:
This will enable Ethernet driver to be used when compiling for native-sim board.

jukkar committed:
drivers: eth: native: Add native-sim support to posix Ethernet driver

This will enable Ethernet driver to be used when compiling
for native-sim board.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

aescolar commented on drivers/ethernet/Kconfig.native_posix:
```diff
@@ -5,7 +5,7 @@
 menuconfig ETH_NATIVE_POSIX
 	bool "Native Posix Ethernet driver"
-	depends on ARCH_POSIX && EXTERNAL_LIBC
+	depends on ARCH_POSIX && (BOARD_NATIVE_POSIX || BOARD_NATIVE_SIM)
```
I guess the new extra checks are not really needed?

jukkar committed:
drivers: eth: native: Remove automatic start script support

As the automatic start requires that the zephyr exe is run as
a root user, it is somewhat cumbersome to use. Remove the script
support and require that the zeth device is created beforehand
by the net-setup.sh script found in net-tools zephyr project.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
samples: net: Remove eth_native_posix sample app

This application is no longer needed as we have removed
the automatic startup support which was used by this sample.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
board: native: Update the posix Ethernet driver documentation

Remove the mention of startup script and update the documentation
to reflect the current way of working i.e., using net-setup.sh script
before starting the zephyr.exe process.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>
